src/main/java/plugily/projects/villagedefense/events/spectator/SpectatorItemEvents.java

```diff
@@ -79,30 +78,29 @@ public void onSpectatorItemClick(CBPlayerInteractEvent e) {
     }
   }
 
-  private void openSpectatorMenu(World world, Player player, Arena arena) {
+  private void openSpectatorMenu(Player player, Arena arena) {
     int rows = Utils.serializeInt(arena.getPlayers().size()) / 9;
     ChestGui gui = new ChestGui(rows, plugin.getChatManager().colorMessage(Messages.SPECTATOR_MENU_NAME));
     OutlinePane pane = new OutlinePane(9, rows);
     gui.addPane(pane);
 
-    ItemStack skull = XMaterial.PLAYER_HEAD.parseItem();
-    SkullMeta meta = (SkullMeta) skull.getItemMeta();
-
-    for(Player arenaPlayer : world.getPlayers()) {
-      if(arena.getPlayers().contains(arenaPlayer) && !plugin.getUserManager().getUser(arenaPlayer).isSpectator()) {
-        meta = VersionUtils.setPlayerHead(arenaPlayer, meta);
-        ComplementAccessor.getComplement().setDisplayName(meta, arenaPlayer.getName());
-        ComplementAccessor.getComplement().setLore(meta, Collections.singletonList(plugin.getChatManager().colorMessage(Messages.SPECTATOR_TARGET_PLAYER_HEALTH)
-            .replace("%health%", Double.toString(NumberUtils.round(arenaPlayer.getHealth(), 2)))));
-        skull.setItemMeta(meta);
-        pane.addItem(new GuiItem(skull, e -> {
-          e.setCancelled(true);
-          e.getWhoClicked().sendMessage(plugin.getChatManager().formatMessage(arena, plugin.getChatManager().colorMessage(Messages.KITS_TELEPORTER_TELEPORTED_TO_PLAYER), arenaPlayer));
-          e.getWhoClicked().closeInventory();
-          e.getWhoClicked().teleport(arenaPlayer);
-        }));
+    for(Player arenaPlayer : arena.getPlayers()) {
+      if(plugin.getUserManager().getUser(arenaPlayer).isSpectator()) {
+        continue;
       }
+      ItemStack skull = XMaterial.PLAYER_HEAD.parseItem();
```

Comment:
Move this outside of for loop.

Reply:
I'll move it... but we have to clone it as the display item will be updated when we change the meta